gateway/handler.go

```diff
@@ -88,9 +87,13 @@ func NewHandler(
 	dedupCleaner *dedup.Cleaner,
 	fallbackURL *url.URL,
 ) http.Handler {
-	var fallbackProxy *gohttputil.ReverseProxy
+	var fallbackHandler http.Handler
 	if fallbackURL != nil {
-		fallbackProxy = gohttputil.NewSingleHostReverseProxy(fallbackURL)
+		fallbackProxy := gohttputil.NewSingleHostReverseProxy(fallbackURL)
+		fallbackHandler = http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			request.Host = strings.Replace(request.Host, bareDomain, fallbackURL.Host, 1)
+			fallbackProxy.ServeHTTP(writer, request)
+		})
```

Comment:
We model this one as a handler after the repo enrich - in case there is no repo it will call the proxy (if configured)